### PR TITLE
Add draft doc for model interface

### LINKED ISSUE #7060: [DOC] Add documentation for model interface
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

To make it easier for user to learn what's the ml-commons model interface and other components to check model interface easily, we will support specifying model interface (input/output parameter list and type) when create model. Read this Github issue for more details https://github.com/opensearch-project/ml-commons/issues/2354

PR : https://github.com/opensearch-project/ml-commons/pull/2357

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by b4sjoo**
### Description
This PR should add draft doc for model interface

### Issues Resolved
Closes #7060 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 1]
```diff
@@ -304,12 +313,100 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
-  "status" : "CREATED"
+    "task_id": "tsqvDY4BChVAiNVEuo8F",
+    "status": "CREATED",
+    "model_id": "t8qvDY4BChVAiNVEuo8q"
+}
+```
+
+### The `interface` parameter
+
+Interface introduces a highly flexible way for adding arbitrary metadata annotations to all local DL models and remote models in JSON schema syntax. This annotation will initiate a validation check on the input and output fields of the model during model invocation. The validation check will ensure that the input and output fields are in the correct format before and after the model performs.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/register-model.md:328]:
```suggestion
The model interface provides a highly flexible way for adding arbitrary metadata annotations to all local deep-learning models and remote models in a JSON schema syntax. This annotation initiates a validation check on the input and output fields of the model during the model's invocation. The validation check ensures that the input and output fields are in the correct format before and after the model performs a prediction.
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 2]
```diff
@@ -304,12 +313,100 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+
+To register a model with model interface, provide the `interface` parameter, which supports the following fields.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/register-model.md:330]:
```suggestion
To register a model with a model interface, provide the `interface` parameter, which supports the following fields. 
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 3]
```diff
@@ -304,12 +313,100 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+}
+```
+
+### The `interface` parameter
+
+Interface introduces a highly flexible way for adding arbitrary metadata annotations to all local DL models and remote models in JSON schema syntax. This annotation will initiate a validation check on the input and output fields of the model during model invocation. The validation check will ensure that the input and output fields are in the correct format before and after the model performs.
+
+To register a model with model interface, provide the `interface` parameter, which supports the following fields. 
+
+Input and output fields will be evaluated against the JSON schema provided separately. User does not necessarily need to provide both input and output fields simultaneously.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/register-model.md:332]: Let's move lines 332 and 334 underneath the table.

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 4]
```diff
@@ -304,12 +313,100 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+
+User can check [JSON schema syntax doc](https://json-schema.org/understanding-json-schema/) to get more information about JSON schema syntax.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/register-model.md:334]:
```suggestion
To learn more about the JSON schema syntax, see [Understanding JSON schema](https://json-schema.org/understanding-json-schema/).
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _ml-commons-plugin/api/model-apis/update-model.md [hunk 5]
```diff
@@ -37,6 +37,7 @@ Field | Data type |  Description
 `rate_limiter.limit` | Integer | The maximum number of times any user can call the Predict API on the model per `unit` of time. By default, there is no limit on the number of Predict API calls. Once you set a limit, you cannot reset it to no limit. As an alternative, you can specify a high limit value and a small time unit, for example, 1 request per nanosecond.
 `rate_limiter.unit` | String | The unit of time for the rate limiter. Valid values are `DAYS`, `HOURS`, `MICROSECONDS`, `MILLISECONDS`, `MINUTES`, `NANOSECONDS`, and `SECONDS`.
 `guardrails`| Object | The guardrails for the model.
+`interface`| Object | Optional | The interface for the model.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/update-model.md:40]: The other rows has 3 columns, this row has 4 columns. Is that ok ?

Comment (b4sjoo) [reply _ml-commons-plugin/api/model-apis/update-model.md]: Fixed

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 6]
```diff
@@ -90,7 +92,9 @@ Field | Data type | Required/Optional | Description
 `url` | String | Required | The URL that contains the model. |
 `description` | String | Optional| The model description. |
 `model_group_id` | String | Optional | The model group ID of the model group to register this model to. 
-`is_enabled`| Boolean | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
+`is_enabled`| Boolean | Optional | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
+`rate_limiter` | Object | Optional | Limits the number of times any user can call the Predict API on the model. For more information, see [Rate limiting inference calls]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#rate-limiting-inference-calls).
+`interface`| Object | Optional | The interface for the model. For more information, see [Interface](#the-interface-parameter).|
```
[the remark this answers is not in the document]
Comment (b4sjoo) [reply _ml-commons-plugin/api/model-apis/register-model.md]: Pre-trained you can only do update, fixed in register part.

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 7]
```diff
@@ -56,8 +56,7 @@ Field | Data type | Required/Optional | Description
 `version` | String | Required | The model version. |
 `model_format` | String | Required | The portable format of the model file. Valid values are `TORCH_SCRIPT` and `ONNX`. |
 `description` | String | Optional| The model description. |
-`model_group_id` | String | Optional | The model group ID of the model group to register this model to. 
-`is_enabled`| Boolean | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/register-model.md:60]: I see you removed this line from pretrained model. Does it mean user can't set is_enabled for pretrained model ?

**Review by Naarcha-AWS — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 8]
```diff
@@ -56,8 +56,7 @@ Field | Data type | Required/Optional | Description
...
+`model_group_id` | String | Optional | The model group ID of the model group to register this model to.
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:59]:
```suggestion
`model_group_id` | String | Optional | The ID of the model group to which to register the model.
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 9]
```diff
@@ -89,8 +88,7 @@ Field | Data type | Required/Optional | Description
 `model_content_hash_value` | String | Required | The model content hash generated using the SHA-256 hashing algorithm.
 `url` | String | Required | The URL that contains the model. |
 `description` | String | Optional| The model description. |
-`model_group_id` | String | Optional | The model group ID of the model group to register this model to. 
-`is_enabled`| Boolean | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
+`model_group_id` | String | Optional | The model group ID of the model group to register this model to.
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:91]:
```suggestion
`model_group_id` | String | Optional | The ID of the model group to which to register this model.
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 10]
```diff
@@ -124,7 +122,9 @@ Field | Data type | Required/Optional | Description
 `url` | String | Required | The URL that contains the model. |
 `description` | String | Optional| The model description. |
 `model_group_id` | String | Optional | The model group ID of the model group to register this model to. 
-`is_enabled`| Boolean | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
+`is_enabled`| Boolean | Optional | Specifies whether the model is enabled. Disabling the model makes it unavailable for Predict API requests, regardless of the model's deployment status. Default is `true`.
+`rate_limiter` | Object | Optional | Limits the number of times any user can call the Predict API on the model. For more information, see [Rate limiting inference calls]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#rate-limiting-inference-calls).
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:126]:
```suggestion
`rate_limiter` | Object | Optional | Limits the number of times that any user can call the Predict API on the model. For more information, see [Rate limiting inference calls]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#rate-limiting-inference-calls).
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 11]
```diff
@@ -182,8 +182,10 @@ Field | Data type | Required/Optional | Description
...
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:186]:
```suggestion
`rate_limiter` | Object | Optional | Limits the number of times that any user can call the Predict API on the model. For more information, see [Rate limiting inference calls]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#rate-limiting-inference-calls).
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 12]
```diff
@@ -304,12 +307,99 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+{
+    "task_id": "tsqvDY4BChVAiNVEuo8F",
+    "status": "CREATED",
+    "model_id": "t8qvDY4BChVAiNVEuo8q"
+}
+```
+
+### The `interface` parameter
+
+The model interface provides a highly flexible way for adding arbitrary metadata annotations to all local deep-learning models and remote models in a JSON schema syntax. This annotation initiates a validation check on the input and output fields of the model during the model's invocation. The validation check ensures that the input and output fields are in the correct format before and after the model performs a prediction.
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:322]:
```suggestion
The model interface provides a highly flexible way to add arbitrary metadata annotations to all local deep learning models and remote models in a JSON schema syntax. This annotation initiates a validation check on the input and output fields of the model during the model's invocation. The validation check ensures that the input and output fields are in the correct format both before and after the model performs a prediction.
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 13]
```diff
@@ -304,12 +307,99 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+To register a model with a model interface, provide the `interface` parameter, which supports the following fields.
+
+Field | Data type | Description                         
+:---  | :--- |:------------------------------------
+`input`| Object | The JSON schema for the model input. |
+`output`| Object | The JSON schema for the model output. |
+
+Input and output fields will be evaluated against the JSON schema provided separately. User does not necessarily need to provide both input and output fields simultaneously.
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:330]:
```suggestion
The input and output fields will be evaluated against the separately provided JSON schema. You do not necessarily need to provide both input and output fields simultaneously.
```

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 14]
```diff
@@ -304,12 +307,99 @@ For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons
...
+
+To learn more about the JSON schema syntax, see [Understanding JSON schema](https://json-schema.org/understanding-json-schema/).
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:332]:
```suggestion
To learn more about the JSON schema syntax, see [Understanding JSON Schema](https://json-schema.org/understanding-json-schema/).
```

**Review by natebower — COMMENTED** [review]
@b4sjoo @Naarcha-AWS Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**